extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -93,7 +104,61 @@ class InstrumentationService {
           }
         });
       }
+    } else if (eventType === AnalyticsEventType.CLICK) {
+      this.ensureClickListener_();
+      this.clickObservable_.add(
+          this.createSelectiveListener_(listener, opt_selector));
+    }
+  }
+
+  ensureClickListener_() {
+    if (!this.clickHandlerRegistered_) {
+      this.clickHandlerRegistered_ = true;
+      this.win_.document.documentElement.addEventListener(
+          'click', this.onClick_.bind(this));
+    }
+  }
+
+  /**
+   * @param {!Event} e
+   */
+  onClick_(e) {
+    this.clickObservable_.fire(e);
+  }
+
+  /**
+   * @param {!Function} listener
+   * @param {string=} opt_selector
+   */
+  createSelectiveListener_(listener, opt_selector) {
+    const wrappedListener = e => {
+      if (opt_selector) {
+        if (!this.matchesSelector_(e.target, opt_selector)) {
+          return;
+        }
+      }
+      listener(new AnalyticsEvent(AnalyticsEventType.CLICK));
+    };
+    return e => {
+      vsyncFor(this.win_).measure(() => {wrappedListener(e);});
+    };
+  }
+
+  /**
+   * @param {Element} el
+   * @param {string} selector
+   * @return {boolean} True if the given element matches the given selector.
+   */
+  matchesSelector_(el, selector) {
+    try {
+      const matches = this.win_.document.querySelectorAll(selector);
```

Comment:
Can we use instead `element.matches(selector)`? Otherwise `querySelectorAll` is super expensive. https://developer.mozilla.org/en-US/docs/Web/API/Element/matches

I know that there was some confusion whether it's `matches` or `matchesSelector` naming. But I believe that `matches` is standard. We may also need to look for vendor prefixes, such as `mozMatches` or `mozMatchesSelector`. I think it's ok to polyfill this method here kind of like you do now, but only after we check all other methods - i.e. the last option.